### DISCUSSION TITLE
Proposition - Custom unity inspectors to hide irrelevant settings

### DISCUSSION THREAD

**Discussion by kipash** [needle-tools/needle-engine-support#72] [Feature Ideas]
Common components like MeshRenderers or Camera could have a custom editor which would hide the unused features and potentially could hardcode states for unused features.

For an example: Disable "occlusion culling" since this setting is not considered in the conversion.

Potentially the original editor could be rendered in a foldout under it, just for clarity.

Code of a custom inspector which renders the original inspector made by unity AND has the ability to render anything before and behind it.
https://gist.github.com/ChrissTman/bebf9fefd525019454235ceb1f6fb020

Obviously, the custom editor would have to check for the presence of the Needle component in the scene which hopefully wouldn't be too expensive.

Might be a nice streamline and would be simpler to understand supported and unsupported unity/needle features.

**Comment by ROBYER1** [comment]
Just to add to this, I was going to write an issue to request the dynamic occlusion culling options for SkinnedMeshRenderer and MeshRenderer are added to the export as frustum culling on/off in Three.js
